sdktests/helpers.go

```diff
@@ -42,6 +46,46 @@ func basicEvaluateFlag(
 	return result.Value
 }
 
+// computeExpectedBucketValue implements the bucketing hash value calculation as per the evaluation spec,
+// except that it returns the value as an integer in the range [0, 100000] - currently the SDKs convert
+// this to a floating-point fraction by in effect dividing it by 100000, but this test code needs an
+// integer in order to compute bucket weights.
+func computeExpectedBucketValue(
+	userValue ldvalue.Value,
+	flagOrSegmentKey, salt string,
+	secondary ldvalue.OptionalString,
+	seed ldvalue.OptionalInt,
+) int {
+	hashInput := ""
+
+	if seed.IsDefined() {
+		hashInput += strconv.Itoa(seed.IntValue())
+	} else {
+		hashInput += flagOrSegmentKey + "." + salt
+	}
+	if !userValue.IsString() {
+		if !userValue.IsInt() {
+			return 0
+		}
+		userValue = ldvalue.String(strconv.Itoa(userValue.IntValue()))
+	}
+	hashInput += "." + userValue.StringValue()
+	if secondary.IsDefined() {
+		hashInput += "." + secondary.StringValue()
+	}
+
+	hashOutputBytes := sha1.Sum([]byte(hashInput)) //nolint:gosec // this isn't for authentication
+	hexEncodedChars := make([]byte, 64)
+	hex.Encode(hexEncodedChars, hashOutputBytes[:])
+	hash := hexEncodedChars[:15]
+
+	hashVal, _ := strconv.ParseInt(string(hash), 16, 64)
+	var product, result big.Int
+	product.Mul(big.NewInt(hashVal), big.NewInt(100000))
+	result.Div(&product, big.NewInt(0xFFFFFFFFFFFFFFF))
```

Comment:
We've had an ongoing discussion for quite a while about whether it would be best for the SDKs to use integer math instead of floating point in the hash calculation. This is the first time we actually tried doing that— and I realized that of course it can't be done with plain uint64s, because `(hashVal * 100000) / 0xFFFFFFFFFFFFFFF` will overflow in the multiplication step. The quickest solution here was to use Go's `big` types. We can discuss this further within the team if necessary in terms of whether it'd be worthwhile to do anything similar in the actual SDK code.